sql/2020/08_Accessibility/skip_links.sql

```diff
@@ -0,0 +1,34 @@
+#standardSQL
+# % of pages having skip links
+CREATE TEMPORARY FUNCTION getEarlyHash(payload STRING)
+RETURNS INT64 LANGUAGE js AS '''
+try {
+  const almanac = JSON.parse(payload);
+  return almanac['seo-anchor-elements'].earlyHash;
+} catch (e) {
+  return 0;
+}
+''';
+
+SELECT
+  _TABLE_SUFFIX AS client,
+  COUNT(DISTINCT url) AS pages,
+  total,
+  COUNT(DISTINCT url) / total AS pct
+FROM
+  `httparchive.pages.2020_08_01_*`
+JOIN (
+  SELECT
+    _TABLE_SUFFIX,
+    COUNT(0) AS total
+  FROM
+    `httparchive.pages.2020_08_01_*`
+  GROUP BY
+    _TABLE_SUFFIX
+)
+USING (_TABLE_SUFFIX)
+WHERE
+  getEarlyHash(JSON_EXTRACT_SCALAR(payload, '$._almanac')) > 0
+GROUP BY
+  client,
+  total
```

Comment:
Same results, just a bit cleaner
```suggestion
SELECT
  _TABLE_SUFFIX AS client,
  COUNTIF(getEarlyHash(JSON_EXTRACT_SCALAR(payload, '$._almanac')) > 0) AS pages,
  COUNT(0) AS total,
  COUNTIF(getEarlyHash(JSON_EXTRACT_SCALAR(payload, '$._almanac')) > 0) / COUNT(0) AS pct
FROM
  `httparchive.pages.2020_08_01_*`
GROUP BY
  client
```

Reply:
LGTM.

I do wonder how this affects the performance of the query. Ideally BQ would be able to optimize this only running the `getEarlyHash` function once... but I guess it has no way to know if the function is idempotent

Reply:
Both queries run in about 45 seconds and consume the same amount of data. It seems BQ is able to optimize the repeated function calls.

Reply:
Wow thats really cool